packages/ssz/src/viewDU/container.ts

```diff
@@ -114,6 +114,44 @@ class ContainerTreeViewDU<Fields extends Record<string, Type<unknown>>> extends
     // However preserving _SOME_ caches results in a very unpredictable experience.
     this.viewsChanged.clear();
   }
+
+  /**
+   * Same method to `type/container.ts` that call ViewDU.serializeToBytes() of internal fields.
+   */
+  serializeToBytes(output: ByteViews, offset: number): number {
+    this.commit();
+
+    let fixedIndex = offset;
+    let variableIndex = offset + this.type.fixedEnd;
+    for (let index = 0; index < this.type.fieldsEntries.length; index++) {
+      const {fieldType} = this.type.fieldsEntries[index];
+      let node = this.nodes[index];
+      if (node === undefined) {
+        node = getNodeAtDepth(this._rootNode, this.type.depth, index);
+        this.nodes[index] = node;
+      }
+      if (fieldType.fixedSize === null) {
+        // write offset
+        output.dataView.setUint32(fixedIndex, variableIndex - offset, true);
+        fixedIndex += 4;
+        // write serialized element to variable section
+        // basic types always have fixedSize
+        if (isCompositeType(fieldType)) {
```

Comment:
do we have some test case(s) covering all the if conditions here?

Reply:
yeah the existing test cases implicitly covered this but it's good to have an explicit test case for it, I added it. `runViewTestMutation` compares `type.serialize()` vs `ViewDU.serialize()`
also spec test with `eth2` types scanned through the code too